jpa-criteria/api/src/main/java/com/blazebit/persistence/criteria/BlazeCriteriaBuilder.java

```diff
@@ -638,4 +638,6 @@ public interface BlazeCriteriaBuilder extends CriteriaBuilder {
      * @return the ordered set-aggregate function expression
      */
     public BlazeOrderedSetAggregateFunctionExpression<String> listaggDistinct(Expression<String> expression, Expression<String> separator);
+
+    public <T> BlazeCriteriaSetQuery<T> union(BlazeCriteriaQuery<T> a, BlazeCriteriaQuery<T> b);
```

Comment:
This is where we might need an abstraction between `BlazeCriteriaQuery` and `BlazeCriteriaSetQuery` in order to support nested sets.

Reply:
In Hibernate, we took the path of letting `CriteriaQuery` just wrap `SqmQueryPart` which can be either a `SqmQuerySpec` or a `SqmQueryGroup`. It delegates all select statement methods to the `SqmQuerySpec` and throws an exception if the thing is not a `SqmQuerySpec`. I think I like that approach and we might be able to pull that off by introducing an abstract class or interface between `InternalQuery` and `InternalSetQuery`. Wdyt?